plugins/inputs/vsphere/endpoint.go

```diff
@@ -1338,6 +1341,9 @@ func (e *Endpoint) populateTags(objectRef *objectRef, resourceType string, resou
 	if resourceType == "vm" && objectRef.rpname != "" {
 		t["rpname"] = objectRef.rpname
 	}
+	if resourceType == "vm" && objectRef.memorySizeMB != 0 {
+		t["memorySizeMB"] = strconv.Itoa(int(objectRef.memorySizeMB))
+	}
```

Comment:
I believe the original concern was that memory size, a numeric value, should not be a tag. Happy to see this as a field, but setting as a tag would cause a lot of issues with cardinality.